### ISSUE TITLE
Functionbeat fails to re-bootstrap ILM index after index deletion

### ISSUE THREAD

**Issue by ppf2** [elastic/beats#19313]
We have a lambda function that writes to Cloudwatch and is using functionbeat (cloudwatch_logs_kinesis) type.  

When the lambda function was run after an initial functionbeat deployment, everything worked fine, i.e., it created the first index `upgrade-fb-version-<date_pattern>-000001 ` and it also created the write alias for that index.  See how it detected the license and confirms that the cluster has ILM enabled, etc..

![image](https://user-images.githubusercontent.com/7216393/84950136-40b54200-b0a3-11ea-94b4-f103d7b5f8a8.png)

(btw, the jsontransform errors are expected and it is present when things are working as well, so it shouldn't affect how functionbeat bootstrap the first index).

However, if we deleted the `upgrade-fb-version-<date_pattern>-000001 ` index and then re-ran our lambda job to generate more logs.  When functionbeat picked up the new logs and created the index again, it failed to bootstrap the ILM index.  In fact, it looks like it never ran the license check again. 

![image](https://user-images.githubusercontent.com/7216393/84950428-be794d80-b0a3-11ea-8fc5-c579a51a12db.png)

As a result, functionbeat ended up creating an index named `upgrade-fb-version` that is a non-ILM managed index:

![image](https://user-images.githubusercontent.com/7216393/84950539-ef598280-b0a3-11ea-8657-fc66abf477b2.png)
 
After a fresh deploy of functionbeat and the successfully initial bootstrap of the ILM index, we waited 60m before deleting the index.  After that, functionbeat is _able_ to re-boostrap the index correctly again without any change in the setup.

In other words, functionbeat appears to have trouble re-bootstrapping the index if the index is deleted "too soon".  We have these ILM/template related settings in the yml:

```
setup.ilm.policy_name: test_default_ilm_policy
setup.ilm.rollover_alias: upgrade-fb-version
# not necessary, but added in an attempt to force functionbeat to honor ILM setup, pattern and custom template
setup.ilm.pattern: "{now/d}-000001" 
setup.ilm.enabled: true
setup.template.enabled: false
```

This is likely related to https://github.com/elastic/beats/issues/17607.   Discussed with @ycombinator , we are filing a separate issue here for functionbeat.
